packages/frontend/src/state/positions/providers.tsx

```diff
@@ -95,15 +95,19 @@ export const ComputeSwapsProvider: FC = ({ children }) => {
     [isWethToken0, data?.swaps, getUsdAmt],
   )
 
-  useEffect(() => {
+  useAppEffect(() => {
     if (computedSwaps.squeethAmount.isGreaterThan(0) && oSqueethBal?.isGreaterThan(0)) {
       setPositionType(PositionType.LONG)
     } else if (computedSwaps.squeethAmount.isLessThan(0)) {
       setPositionType(PositionType.SHORT)
     } else setPositionType(PositionType.NONE)
   }, [computedSwaps.squeethAmount, oSqueethBal, setPositionType])
 
-  const value = useMemo(
+  useAppEffect(() => {
+    refetch()
+  }, [computedSwaps.squeethAmount, refetch])
```

Comment:
Do we include oSqueethBal in the dep array?

Reply:
I don't think we should include it as it'll cause infinite rerendering.

Reply:
oh yh it will 😂